flips/20211007-transaction-fees.md

```diff
@@ -55,151 +57,152 @@ From the users perspective the transaction fees implementation should satisfy th
+In the second scenario the payer is charged for the transaction fees with the execution effort set to 0.
 
-In the third case, the fees are charged normally with the dynamic computation effort set to the actual usage before the error was produced.
+In the third case, the fees are charged normally with the execution effort set to the actual usage before the error was produced.
 
-Finally in the last case the payer is charged with the transaction fees where the dynamic computation effort is set to the computation limit.
+Finally in the last case the payer is charged with the transaction fees where the execution effort is set to the execution effort limit.
 
 ### Preventing malicious actors
 
 The purpose of fees is to prevent malicious actors sending transactions with the purpose of overloading the network. To achieve that, we have to check that the payer of a transaction has the funds to pay for that transaction as soon as possible.
 
-Tho achieve this the following checks should happen on the collection node, before the transaction is even included in a collection: 
+Tho achieve this the following checks should happen on the access node, before the transaction is even included in a collection: 
 - Is the payers signature valid.
-- Can the payer pay for the transaction. The payers default FLOW vault should have at least <img src="https://render.githubusercontent.com/render/math?math=F%5E%5Ctext%7Bmax%7D"> FLOW. <img src="https://render.githubusercontent.com/render/math?math=F%5E%5Ctext%7Bmax%7D%20%3D%20F_i%20%2B%20(C_c%20%2B%20D_c%5E%5Ctext%7Bmax%7D)%20f_c"> 
+- Can the payer pay for the transaction. The payers default FLOW vault should have at least <img src="https://render.githubusercontent.com/render/math?math=F%5E%5Ctext%7Bmax%7D"> FLOW. <img src="https://render.githubusercontent.com/render/math?math=F%5E%7Bmax%7D%20%3D%20f%20(%20E_i%20%2B%20E_e%5E%7Bmax%7D%20)">
 
-It might still happen that this transaction will get included, either due to a bug or malicious behaviour from the collection node, or due to the collection node not having up to date information of the balance of the payer. In case does still get included in a collection, even though it does not meet the condition, this check should be repeated in the execution node, and in case the check fails on the execution node, the transaction fees should be deducted from the collection nodes account instead of the payers account.
+It might still happen that this transaction will get included in a collection, either due to a bug or malicious behaviour from the access node, or due to the access node not having up to date information of the balance of the payer. In case does still get included in a collection, even though it does not meet the condition, this check should be repeated in the execution node, and in case the check fails on the execution node, the transaction fees should be deducted from the access nodes account instead of the payers account.
```

Comment:
I think we should be explicitly calling out that with this we propose that access nodes will need to handle the scenario where an account attacks the access node by submitting transaction that will drain the access node funds - for example by having the ability to charge the account that submitted the transaction via alternative payment methods.

Reply:
I added more text about this, but the issue is still open ended.